### PR TITLE
#170947436 login with facebook and google

### PR DESCRIPTION
**Pull request by blaise82**
#### What does this PR do? 
implements Facebook and Google login 

#### Description of Task to be completed?
- create user routes
- create passport config
- create user controller

#### How should this be manually tested?
- Clone this repo
- Checkout to branch ft-facebook-goole-login-170947436
- Run npm install
- Add environment variables to .env **(if you don't have a .env file please create it  octopus-bn-backend/.env)**
FACEBOOK_APP_ID=' '              ||  [How to get Facebook application id and secret key?](https://theonetechnologies.com/blog/post/how-to-get-facebook-application-id-and-secret-key)
FACEBOOK_APP_SECRET=' '   || [How to get Facebook application id and secret key?](https://theonetechnologies.com/blog/post/how-to-get-facebook-application-id-and-secret-key)
GOOGLE_APP_ID=' '                  || [How to get Google app Client ID and Client Secret](https://theonetechnologies.com/blog/post/how-to-get-google-app-client-id-and-client-secret)
GOOGLE_APP_SECRET=' '       || [How to get Google app Client ID and Client Secret](https://theonetechnologies.com/blog/post/how-to-get-google-app-client-id-and-client-secret)
- Excuting 
       - you can use the following postman collection https://documenter.getpostman.com/view/9143901/SWTK4twu
        or - use the following routes 
                    - Google: localhost:3000/api/v1/auth/google
                    - Google: localhost:3000/api/v1/auth/facebook
- Provide access token 
in your broswer navigate to 

    - Google -- `http://localhost:3000/api/v1/auth/google`     
    - Facebook -- `http://localhost:3000/api/v1/auth/facebok  `
#### Screenshots (if appropriate)

- Facebook screenshot 
Screen Shot 2020-02-12 at 13 31 04

- Google screenshot
Screen Shot 2020-02-12 at 13 32 55

- database view
Screen Shot 2020-02-11 at 10 35 40

#### Questions:
Fixes #

[finishes #170947436]

### CODE REVIEW

**Review by izzett222 — APPROVED** [review]

File: src/config/passport.js [hunk 1]
```diff
@@ -1,26 +1,35 @@
...
+
+dotenv.config();
+
+
+passport.serializeUser((user, done) => done(null, user));
+passport.deserializeUser((user, done) => done(null, user));
+// Use the GoogleStrategy within Passport.
+//   Strategies in Passport require a `verify` function, which accept
+//   credentials (in this case, an accessToken, refreshToken, and Google
+//   profile), and invoke a callback with a user object.
```
Comment (hakoemmy) [inline src/config/passport.js]: Would you please remove comment lines?

**Review by hakoemmy — CHANGES_REQUESTED** [review]

File: src/controllers/user.controller.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        where: { email }, // we search for this user
+        defaults: {
+          userID,
+          firstName,
+          lastName,
+          method,
+          email,
+          isVerified
+        } // if it doesn't exist, we create it with this additional data
+      });
```
Comment (hakoemmy) [inline src/controllers/user.controller.js]: Get rid of these comments, please

**Review by hakoemmy — CHANGES_REQUESTED** [review]

File: src/index.js [hunk 3]
```diff
@@ -2,13 +2,30 @@ import express from 'express';
...
+import methodOverride from 'method-override';
+import morgan from 'morgan';
 import errorhandler from 'errorhandler';
+import cookieSession from 'cookie-session';
 import router from './routes/index';
 
 const isProduction = process.env.NODE_ENV === 'production';
 
 // Create global app object
 const app = express();
```
Comment (hakoemmy) [inline src/index.js:18]: Would you please get rid of this comment

**Review by hakoemmy — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 3 above)
Comment (hakoemmy) [inline src/index.js:18]: Would you please remove this comment?

**Review by hakoemmy — CHANGES_REQUESTED** [review]

**Review by hakoemmy — CHANGES_REQUESTED** [review]
@blaise82 Great work. But endeavour to get rid of all comments in your codebase.

File: src/config/passport.js (hunk 1 above)
Comment (blaise82) [reply src/config/passport.js]: on it

Comment (hakoemmy) [reply src/config/passport.js]: @blaise82 Good to know

Comment (blaise82) [reply src/config/passport.js]: done @hakoemmy

File: src/index.js (hunk 3 above)
Comment (blaise82) [reply src/index.js:18]: @hakoemmy  some comments were there even before 
so I don't think i am allowed to remove them 

for example, the one you're suggesting me to remove

Comment (hakoemmy) [reply src/index.js:18]: Ok. @blaise82 Remove yours only

**Review by hakoemmy — APPROVED** [review]

**Review by RusimbiPatrick — APPROVED** [review]

**Review by AbdoulNuru — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -6,7 +6,8 @@
   "scripts": {
     "start": "node build/index.js",
     "dev": "nodemon --exec babel-node src/index.js",
-    "test": "nyc --reporter=html --reporter=text mocha  --require @babel/register 'src/**/**test.js' --exit",
+    "test": "NODE_ENV=test npm run migrateTest && NODE_ENV=test nyc --reporter=html --reporter=text mocha  --require @babel/register 'src/**/**test.js'  --timeout 200000 --exit",
```
Comment (kcharles52) [inline package.json:9]:
```suggestion
    "test": "NODE_ENV=test npm run migrateTest && nyc --reporter=html --reporter=text mocha  --require @babel/register 'src/**/**test.js'  --timeout 200000 --exit",
```

**Review by kcharles52 — CHANGES_REQUESTED** [review]

File: package.json (hunk 4 above)
Comment (blaise82) [reply package.json:9]: on it

Comment (blaise82) [reply package.json:9]: @kcharles52  changes implemented

**Pull request merged**